### PR TITLE
Make sure log groups are created before and deleted after lambdas

### PR DESCRIPTION
**Pull request by Nuru**
## what

- Make sure log groups are created before and deleted after lambdas
- Update terraform `cloudposse/module-artifact/external` to v0.8.0
- Update tests

## why

- AWS will create a log group if needed, and if it does, it will clash with the one this module is trying to manage. Ensure that AWS never needs to create a log group by making sure this module creates one before creating the lambda and does not delete it until after the lambda is removed.
- Keep current
- Fix broken tests

## references

- Supersedes and closes #52 
- Incorporates #56, Supersedes and closes #62

### CODE REVIEW

File: lambda-rds.tf [hunk 1]
```diff
@@ -106,6 +109,9 @@ resource "aws_lambda_function" "forwarder_rds" {
   }
 
   tags = module.forwarder_rds_label.tags
+
+  # AWS will create the log group if needed. Make sure we create it first.
+  depends_on = [aws_cloudwatch_log_group.forwarder_vpclogs]
```
[the remark this answers is not in the document]
Comment (Nuru) [reply lambda-rds.tf:114]: Yes, it should.

File: lambda-log.tf [hunk 2]
```diff
@@ -111,6 +114,9 @@ resource "aws_lambda_function" "forwarder_log" {
   }
 
   tags = module.forwarder_log_label.tags
+
+  # AWS will create the log group if needed. Make sure we create it first.
+  depends_on = [aws_cloudwatch_log_group.forwarder_vpclogs]
```
[the remark this answers is not in the document]
Comment (Nuru) [reply lambda-log.tf:119]: Yes, I see, but I do not understand why. It was a late-night copy/paste error.

File: lambda-log.tf [hunk 3]
```diff
@@ -55,6 +55,9 @@ resource "aws_iam_role" "lambda_forwarder_log" {
   assume_role_policy   = data.aws_iam_policy_document.assume_role[0].json
   permissions_boundary = var.log_permissions_boundary
   tags                 = module.forwarder_log_label.tags
+
+  # AWS will create the log group if needed. Make sure we create it first.
+  depends_on = [aws_cloudwatch_log_group.forwarder_vpclogs]
```
[the remark this answers is not in the document]
Comment (Nuru) [reply lambda-log.tf:60]: Yes

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**